airflow/providers/databricks/operators/databricks.py

```diff
@@ -844,7 +832,19 @@ def _get_hook(self, caller: str) -> DatabricksHook:
             caller=caller,
         )
 
+    def _setup_and_validate_json(self):
```

Comment:
NIT: can we make it a bit more DRY - seems that we are repeating the logic in two classes - it has "enough" of logic to be susceptible to errors (and there is no coupling introduced by it - those classes are already coupled). Extracting it to a top-level method with operator passed as parameter should do the job - maybe you can also extract the key extraction for loop to a separate method.

Also:

```
        if self.json:
            self.json = normalise_json_content(self.json)
```

could be turned into:

```
    def normalize_json_content():
        if self.json:
            self.json = _normalise_json_content(self.json)
```

Reply:
Sure, I refactored some parts in [this commit](https://github.com/apache/airflow/pull/40471/commits/1cfaf4c58d8522ab7c24dcc636873f6003a1cf9c) but I was not sure about key checks and raising Exceptions.

Currently, I implemented `def _handle_overridden_json_params(operator)` and `def normalise_json_content(operator):` but kept the checks in between those commonly used parts inside the corresponding operator  itself.

Another approach could be checking the instance of the passed `operator` in `_handle_overridden_json_params` and move all that logic to a single function but IMHO it would complicate it.

With the current approach, code is not repeating and operator specific checks are still inside the corresponding operator.

WDYT?

Reply:
nice!